transport/src/main/java/io/netty/channel/oio/AbstractOioMessageChannel.java

```diff
@@ -89,13 +96,9 @@ protected void doRead() {
             if (isOpen()) {
                 unsafe().close(unsafe().voidPromise());
             }
-        } else if (allocHandle.lastBytesRead() == 0 && isActive()) {
-            // If the read amount was 0 and the channel is still active we need to trigger a new read()
-            // as otherwise we will never try to read again and the user will never know.
-            // Just call read() is ok here as it will be submitted to the EventLoop as a task and so we are
-            // able to process the rest of the tasks in the queue first.
-            //
-            // See https://github.com/netty/netty/issues/2404
+        } else if (readPending || config.isAutoRead() || !readData && isActive()) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Change this condition so that it does not always evaluate to "false" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2583)